// @estateflow/web/tailwind.config.js
/** @type {import('tailwindcss').Config} */
export default {
  content: [
    './index.html',
    './src/**/*.{js,ts,jsx,tsx}',
    '../../packages/shared/src/**/*.{js,ts,jsx,tsx}',
  ],
  theme: {
    extend: {
      colors: {
        primary: '#000000',
        'on-secondary-fixed-variant': '#005236',
        'on-error-container': '#93000a',
        'on-surface-variant': '#45464d',
        outline: '#76777d',
        'primary-container': '#131b2e',
        'on-secondary-container': '#00714d',
        'on-primary-container': '#7c839b',
        'on-tertiary': '#ffffff',
        'tertiary-container': '#001a42',
        'surface-tint': '#565e74',
        'on-secondary-fixed': '#002113',
        'on-primary': '#ffffff',
        'surface-container-low': '#eff4ff',
        tertiary: '#000000',
        'on-primary-fixed': '#131b2e',
        'error-container': '#ffdad6',
        'on-secondary': '#ffffff',
        'tertiary-fixed': '#d8e2ff',
        'surface-container-lowest': '#ffffff',
        surface: '#f8f9ff',
        'surface-dim': '#cbdbf5',
        'inverse-primary': '#bec6e0',
        'on-tertiary-fixed-variant': '#004395',
        background: '#f8f9ff',
        'secondary-container': '#6cf8bb',
        'on-surface': '#0b1c30',
        'on-tertiary-fixed': '#001a42',
        'surface-variant': '#d3e4fe',
        'inverse-surface': '#213145',
        'secondary-fixed': '#6ffbbe',
        'on-background': '#0b1c30',
        'primary-fixed': '#dae2fd',
        'on-tertiary-container': '#3980f4',
        'surface-container-high': '#dce9ff',
        'on-primary-fixed-variant': '#3f465c',
        'inverse-on-surface': '#eaf1ff',
        'tertiary-fixed-dim': '#adc6ff',
        'surface-bright': '#f8f9ff',
        error: '#ba1a1a',
        'primary-fixed-dim': '#bec6e0',
        'surface-container': '#e5eeff',
        'outline-variant': '#c6c6cd',
        secondary: '#006c49',
        'surface-container-highest': '#d3e4fe',
        'on-error': '#ffffff',
        'secondary-fixed-dim': '#4edea3',
      },
      borderRadius: {
        DEFAULT: '0.25rem',
        lg: '0.5rem',
        xl: '0.75rem',
        '2xl': '1rem',
        full: '9999px',
      },
      spacing: {
        'stack-md': '16px',
        'stack-sm': '8px',
        'container-max': '1280px',
        'stack-lg': '32px',
        'margin-desktop': '32px',
        'margin-mobile': '16px',
        gutter: '24px',
        unit: '8px',
      },
      fontFamily: {
        'label-md': ['Inter', 'sans-serif'],
        'label-sm': ['Inter', 'sans-serif'],
        'body-lg': ['Inter', 'sans-serif'],
        'headline-xl': ['Inter', 'sans-serif'],
        'body-sm': ['Inter', 'sans-serif'],
        'headline-lg': ['Inter', 'sans-serif'],
        'body-md': ['Inter', 'sans-serif'],
        'headline-md': ['Inter', 'sans-serif'],
      },
      fontSize: {
        'label-md': ['14px', { lineHeight: '1', letterSpacing: '0.01em', fontWeight: '600' }],
        'label-sm': ['12px', { lineHeight: '1', letterSpacing: '0.02em', fontWeight: '500' }],
        'body-lg': ['18px', { lineHeight: '1.6', fontWeight: '400' }],
        'headline-xl': ['36px', { lineHeight: '1.2', letterSpacing: '-0.02em', fontWeight: '700' }],
        'body-sm': ['14px', { lineHeight: '1.5', fontWeight: '400' }],
        'headline-lg': ['24px', { lineHeight: '1.3', letterSpacing: '-0.01em', fontWeight: '600' }],
        'body-md': ['16px', { lineHeight: '1.5', fontWeight: '400' }],
        'headline-md': ['20px', { lineHeight: '1.4', fontWeight: '600' }],
      },
      maxWidth: {
        'container-max': '1280px',
      },
    },
  },
  plugins: [],
};
